sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -77,6 +80,88 @@
 
   private DoFnInvokers() {}
 
+  /**
+   * Creates a {@link DoFnInvoker} for the given {@link Object}, which should be either a
+   * {@link DoFn} or an {@link OldDoFn}. The expected use would be to deserialize a user's
+   * function as an {@link Object} and then pass it to this method, so there is no need to
+   * statically specify what sort of object it is.
+   *
+   * @deprecated this is to be used only as a migration path for decoupling upgrades
+   */
+  @Deprecated
+  public DoFnInvoker<?, ?> invokerFor(Object deserializedFn) {
+    if (deserializedFn instanceof DoFn) {
+      return newByteBuddyInvoker((DoFn<?, ?>) deserializedFn);
+    } else if (deserializedFn instanceof OldDoFn){
+      return new OldDoFnInvoker<>((OldDoFn<?, ?>) deserializedFn);
+    } else {
+      throw new IllegalArgumentException(String.format(
+          "Cannot create a %s for %s; it should be either a %s or an %s.",
+          DoFnInvoker.class.getSimpleName(),
+          deserializedFn,
```

Comment:
Good point. I want `deserializedFn.toString()`, figuring that it will contain adequate hints as to the class.